File: hilt/app/src/main/java/com/android/hilt/MainActivity.java
package com.android.hilt;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class MainActivity extends AppCompatActivity {
    private final static String TAG = "lcy";
    @Inject
    User user;

    @Inject
    User user1;

    @Inject
    ViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Log.d("TAG", "user: " + user);
        Log.d("TAG", "user1: " + user1);

//        viewModel.test();
        MainViewModel mainViewModel = new ViewModelProvider(this).get(MainViewModel.class);
        mainViewModel.test();

        startActivity(new Intent(this, SecondActivity.class));
    }
}

File: JetpackStudy/viewbinding/src/main/java/com/android/viewbinding/MainActivity.kt
package com.android.viewbinding

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.android.viewbinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding = ActivityMainBinding.inflate(layoutInflater)
        val rootView = binding.root
        setContentView(rootView)


    }
}